mlflow/models/evaluation/base.py

```diff
@@ -204,6 +205,33 @@ def eval_fn(
             )
         name = eval_fn.__name__
 
+    if "/" in name:
+        raise MlflowException(
+            f"Invalid metric name '{name}'. Metric names cannot include forward slashes ('/').",
+            INVALID_PARAMETER_VALUE,
+        )
+
+    if not name.isidentifier():
+        _logger.warning(
+            f"The metric name '{name}' provided is not a valid Python identifier, which will "
+            "prevent its use as a base metric for derived metrics. Please use a valid identifier "
+            "to enable creation of derived metrics that use the given metric."
+        )
+
+    if keyword.iskeyword(name):
+        _logger.warning(
+            f"The metric name '{name}' is a reserved Python keyword, which will "
+            "prevent its use as a base metric for derived metrics. Please use a valid identifier "
+            "to enable creation of derived metrics that use the given metric."
+        )
+
+    if name in ["predictions", "targets", "metrics"]:
+        _logger.warning(
+            f"The metric name '{name}' is used as a special parameter in MLflow metrics, which "
+            "will prevent its use as a base metric for derived metrics. Please use a different "
+            "name to enable creation of derived metrics that use the given metric."
+        )
```

Comment:
why are these warnings instead of errors? it doesnt seem like any of these would actually work.

Reply:
If the metric is not used as part of some dependent/derived metrics, there wouldn't be any issues. For instance, the user could do something like the following

```predictions_metric = make_metric(name="predictions", ...)
quality_metric = make_metric(name="quality-metric", ...)
truth_metric = make_metric(name="True", ...)

results = mlflow.evaluate(..., extra_metrics=[predictions_metric, quality_metric, truth_metric])

assert(results.metrics == {"predictions": ..., "quality-metric": ..., "True": ...})
```

and there wouldn't be any problems